src/wp-includes/class-wp-block.php

```diff
@@ -408,6 +408,22 @@ private function replace_html( string $block_content, string $attribute_name, $s
 	 */
 	public function render( $options = array() ) {
 		global $post;
+
+		/*
+		 * There can be only one root interactive block at a time because the rendered HTML of that block contains
+		 * the rendered HTML of all its inner blocks, including any interactive block.
+		 */
+		static $root_interactive_block = null;
+		if (
+			null === $root_interactive_block &&
+			(
+				( isset( $this->block_type->supports['interactivity'] ) && true === $this->block_type->supports['interactivity'] ) ||
+				! empty( $this->block_type->supports['interactivity']['interactive'] )
+			)
+		) {
+			$root_interactive_block = $this;
```

Comment:
Would be more efficient to add an extra attribute to parsed_block than to compare the entire block?

I remember that the problem comparing $parsed_block attribute in layout has been fixed.

Reply:
Objects are stored and compared by reference, right? How would be any alternative more efficient in that case?

Reply:
Can you share a code example to illustrate the idea?

Reply:
@cbravobernal, I think you're confusing arrays with objects here in PHP. 

The problem lies in arrays, as PHP doesn't compare them by reference, and instead compares the entire array. In this case, checking a property of the array is more performant than checking the entire array. 

However, in the case of objects, they are instances of a class and are compared by reference. So there's no performance problem.

Is that correct, @gziolo?

Reply:
> @cbravobernal, I think you're confusing arrays with objects here in PHP.

Yep, as we were comparing arrays back then in `$parsed_block` if I remember correctly.

Reply:
Yes, but that's no longer the case, right? So we no longer have that problem.